pkg/providers/pricing/pricing.go

```diff
@@ -201,37 +155,10 @@ func (p *Provider) SpotPrice(instanceType string, zone string) (float64, bool) {
 	return 0.0, false
 }
 
-func (p *Provider) updatePricing(ctx context.Context) {
-	var wg sync.WaitGroup
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-		if err := p.UpdateOnDemandPricing(ctx); err != nil {
-			logging.FromContext(ctx).Errorf("updating on-demand pricing, %s, using existing pricing data from %s", err, err.lastUpdateTime.Format(time.RFC3339))
-		}
-	}()
-
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-		if err := p.UpdateSpotPricing(ctx); err != nil {
-			logging.FromContext(ctx).Errorf("updating spot pricing, %s, using existing pricing data from %s", err, err.lastUpdateTime.Format(time.RFC3339))
-		}
-	}()
-
-	wg.Wait()
-}
-
 func (p *Provider) UpdateOnDemandPricing(ctx context.Context) *Err {
-	// standard on-demand instances
-	var wg sync.WaitGroup
-	var onDemandPrices, onDemandMetalPrices map[string]float64
-	var onDemandErr, onDemandMetalErr error
-
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-		onDemandPrices, onDemandErr = p.fetchOnDemandPricing(ctx,
+	filters := map[string][]*pricing.Filter{
```

Comment:
Looking at this now, I think my initial intuition was wrong. This is more obfuscated than the original. Sticking with the waitGroups might be the best call here